src/natcap/invest/urban_nature_access.py

```diff
@@ -459,6 +517,147 @@ def execute(args):
+        target_raster_path (string): Where the reclassified greenspace raster
+            should be written.
+        only_these_greenspace_codes=None (iterable or None): If ``None``, all
+            lucodes with a ``greenspace`` value of 1 will be reclassified to 1.
+            If an iterable, must be an iterable of landuse codes matching codes
+            in then lulc attribute table.  Only these landcover codes will have
+            greenspace area classified in the target raster path.
+
+    Returns:
+        ``None``
+    """
+    attribute_table_dict = utils.build_lookup_from_csv(
+        lulc_attribute_table, key_field='lucode')
+
+    squared_pixel_area = abs(
+        numpy.multiply(*_square_off_pixels(lulc_raster_path)))
+
+    if only_these_greenspace_codes:
+        valid_greenspace_codes = set(only_these_greenspace_codes)
+    else:
+        valid_greenspace_codes = set(
+            lucode for lucode, attributes in attribute_table_dict.items()
+            if (attributes['greenspace']) == 1)
+
+    greenspace_area_map = {}
+    for lucode, attributes in attribute_table_dict.items():
+        greenspace_area = 0
+        if lucode in valid_greenspace_codes:
+            greenspace_area = squared_pixel_area
+        greenspace_area_map[lucode] = greenspace_area
+
+    lulc_raster_info = pygeoprocessing.get_raster_info(lulc_raster_path)
+    greenspace_area_map[lulc_raster_info['nodata'][0]] = FLOAT32_NODATA
+
+    utils.reclassify_raster(
+        raster_path_band=(lulc_raster_path, 1),
+        value_map=greenspace_area_map,
+        target_raster_path=target_raster_path,
+        target_datatype=gdal.GDT_Float32,
+        target_nodata=FLOAT32_NODATA,
+        error_details={
+            'raster_name': ARGS_SPEC['args']['lulc_raster_path']['name'],
+            'column_name': 'greenspace',
+            'table_name': ARGS_SPEC['args']['lulc_attribute_table']['name'],
+        }
+    )
+
+
+def _preprocess_lulc_attribute_table(
+        source_attr_table_path, default_radius, target_attr_table_path):
+    """Preprocess the LULC attribute table.
+
+    This function ensures that we have an attribute table where a radius is
+    defined for every greenspace landcover classification.  For classifications
+    where the radius is not defined, the user-defined default radius is used
+    instead.
+
+    Args:
+        source_attr_table_path (string): The path to a CSV on disk.  Must
+            contain the "lucode" and "greenspace" columns.  May contain a
+            "search_radius_m" column.
+        default_radius (number): The default radius to use when
+            "search_radius_m" is not defined.
+        target_attr_table_path (string): The location on disk where the
+            preprocessed attribute table is written.
+
+    Returns:
+        ``None``
+    """
+    attr_dataframe = utils.read_csv_to_dataframe(
+        source_attr_table_path, to_lower=True, sep=None, engine='python',
```

Comment:
`sep=None` and `engine='python'` are the defaults, but feel free to leave them if you think it helps to be explicit here.

Reply:
I think this function has been removed because this function is no longer needed.